packages/repository/src/decorators/relation.decorator.ts

```diff
@@ -59,14 +69,40 @@ export function hasOne(definition?: Object) {
   return PropertyDecoratorFactory.createDecorator(RELATIONS_KEY, rel);
 }
 
-/**
- * Decorator for hasMany
- * @param definition
- * @returns {(target:any, key:string)}
- */
-export function hasMany(definition?: Object) {
-  const rel = Object.assign({type: RelationType.hasMany}, definition);
-  return PropertyDecoratorFactory.createDecorator(RELATIONS_KEY, rel);
+export function hasMany<T extends typeof Entity>(
+  targetModel: T,
+  definition?: Partial<HasManyDefinition>,
+) {
+  // todo(shimks): extract out common logic (such as @property.array) to
+  // @relation
+  return function(target: Object, key: string) {
+    property.array(targetModel)(target, key);
+
+    const defaultFkName = camelCase(target.constructor.name + '_id');
+    if (
+      !(
+        (definition && definition.keyTo) ||
+        (targetModel.definition &&
+          targetModel.definition.properties &&
+          targetModel.definition.properties[defaultFkName])
```

Comment:
http://loopback.io/doc/en/contrib/style-guide.html#indentation-of-multi-line-expressions-in-if

One of the many possible ways how to improve this long condition:

```ts
const hasKeyTo = definition && definition.keyTo;
const targetPropertyDef = targetModel.definition && targetModel.definition.properties || {};
const hasDefaultFkProperty = targetPropertyDef[defaultFkName);
if (!(hasKeyTo || hasDefaultFkProperty)) {
  // etc.
}
```